pkg/sql/schemachanger/schemachanger_test: TestAlterTableDMLInjection failed

pkg/sql/schemachanger/schemachanger_test.TestAlterTableDMLInjection [failed](https://mesolite.cluster.engflow.com/invocations/default/5cbeb2f6-df05-4c94-9033-cfe733b96f8a?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on release-25.3.0-rc @ [cb540b101146345c1df9f265f65ea321c62b10b3](https://github.com/cockroachdb/cockroach/commits/cb540b101146345c1df9f265f65ea321c62b10b3):


```
=== PAUSE TestAlterTableDMLInjection/drop_column_with_expression_index_cascade
=== CONT  TestAlterTableDMLInjection/drop_column_with_expression_index_cascade
    test_server_shim.go:156: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
		TestAlterTableDMLInjection/drop_column_with_expression_index_cascade (40m0s)
=== RUN   TestAlterTableDMLInjection/validate_constraint
=== PAUSE TestAlterTableDMLInjection/validate_constraint
=== RUN   TestAlterTableDMLInjection/create_expression_index
=== PAUSE TestAlterTableDMLInjection/create_expression_index
=== CONT  TestAlterTableDMLInjection/create_expression_index
    test_server_shim.go:156: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
		TestAlterTableDMLInjection/create_expression_index (40m0s)
=== RUN   TestAlterTableDMLInjection/create_index
=== PAUSE TestAlterTableDMLInjection/create_index
=== RUN   TestAlterTableDMLInjection/create_partial_index
=== PAUSE TestAlterTableDMLInjection/create_partial_index
=== CONT  TestAlterTableDMLInjection/create_partial_index
    test_server_shim.go:156: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
		TestAlterTableDMLInjection/create_partial_index (26m59s)
=== RUN   TestAlterTableDMLInjection/drop_a_column_with_a_check_constraint_while_querying_pg_constraint
=== PAUSE TestAlterTableDMLInjection/drop_a_column_with_a_check_constraint_while_querying_pg_constraint
=== RUN   TestAlterTableDMLInjection/drop_index
=== PAUSE TestAlterTableDMLInjection/drop_index
=== RUN   TestAlterTableDMLInjection/add_column_default_value
=== PAUSE TestAlterTableDMLInjection/add_column_default_value
=== RUN   TestAlterTableDMLInjection/alter_column_type_validate
=== PAUSE TestAlterTableDMLInjection/alter_column_type_validate
=== RUN   TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== PAUSE TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== RUN   TestAlterTableDMLInjection/add_column_default_serial_sql_sequence_cached
=== PAUSE TestAlterTableDMLInjection/add_column_default_serial_sql_sequence_cached
=== RUN   TestAlterTableDMLInjection/drop_column_with_index_using_hash_cascade
=== PAUSE TestAlterTableDMLInjection/drop_column_with_index_using_hash_cascade
=== RUN   TestAlterTableDMLInjection/add_primary_key
=== PAUSE TestAlterTableDMLInjection/add_primary_key
=== RUN   TestAlterTableDMLInjection/alter_column_type_trivial
=== PAUSE TestAlterTableDMLInjection/alter_column_type_trivial
=== RUN   TestAlterTableDMLInjection/add_column_virtual_NOT_NULL
=== PAUSE TestAlterTableDMLInjection/add_column_virtual_NOT_NULL
=== RUN   TestAlterTableDMLInjection/add_constraint_check_udt
=== PAUSE TestAlterTableDMLInjection/add_constraint_check_udt
=== RUN   TestAlterTableDMLInjection/add_constraint_foreign_key
=== PAUSE TestAlterTableDMLInjection/add_constraint_foreign_key
=== CONT  TestAlterTableDMLInjection/add_constraint_foreign_key
    test_server_shim.go:156: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:527: polling for queued jobs to complete: poll-show-jobs: aborted in DistSender: result is ambiguous: node unavailable; try another peer
--- FAIL: TestAlterTableDMLInjection/add_constraint_foreign_key (781.40s)
=== RUN   TestAlterTableDMLInjection/drop_column
=== PAUSE TestAlterTableDMLInjection/drop_column
=== RUN   TestAlterTableDMLInjection/drop_column_stored
=== PAUSE TestAlterTableDMLInjection/drop_column_stored
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableDMLInjection.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53735

--- a/pkg/sql/schemachanger/dml_injection_test.go
+++ b/pkg/sql/schemachanger/dml_injection_test.go
@@ -563,7 +563,6 @@ func TestAlterTableDMLInjection(t *testing.T) {
 	ctx := context.Background()
 	for _, tc := range testCases {
 		t.Run(tc.desc, tc.capture(func(t *testing.T, tc testCase) {
-			t.Parallel() // SAFE FOR TESTING
 			if issue := tc.skipIssue; issue != 0 {
 				skip.WithIssue(t, issue)
 			}

